corehq/apps/domain/views/accounting.py

```diff
@@ -1549,49 +1548,64 @@ class SubscriptionRenewalView(PlanViewBase, SubscriptionMixin):
     step_title = gettext_lazy("Renew Plan")
     template_name = "domain/renew_plan.html"
 
+    @property
+    def current_edition(self):
+        return self.current_subscription.plan_version.plan.edition
+
     @property
     def lead_text(self):
         return format_html(
             _("Based on your current usage we recommend you use the <strong>{plan}</strong> plan"),
-            plan=_(self.current_subscription.plan_version.plan.edition)
+            plan=_(self.current_edition)
         )
 
     @property
     def plan_is_self_renewable(self):
-        return self.subscription.plan_version.plan.edition in SoftwarePlanEdition.SELF_RENEWABLE_EDITIONS
+        return self.current_edition in SoftwarePlanEdition.SELF_RENEWABLE_EDITIONS
+
+    @property
+    @memoized
+    def monthly_plan_version(self):
+        return DefaultProductPlan.get_default_plan_version(
+            edition=self.current_edition, is_annual_plan=False)
 
     @property
     def renewal_choices(self):
-        edition = self.subscription.plan_version.plan.edition
         if not self.plan_is_self_renewable:
             return {}
 
-        monthly_plan = DefaultProductPlan.get_default_plan_version(
-            edition=edition, is_annual_plan=False)
+        monthly_plan = self.monthly_plan_version
         annual_plan = DefaultProductPlan.get_default_plan_version(
-            edition=edition, is_annual_plan=True)
+            edition=self.current_edition, is_annual_plan=True)
         return {'monthly_plan': monthly_plan.user_facing_description,
                 'annual_plan': annual_plan.user_facing_description}
 
+    @property
+    def downgrade_messages(self):
+        downgrades = get_change_status(self.subscription.plan_version, self.monthly_plan_version)[1]
+        downgrade_handler = DomainDowngradeStatusHandler(
+            self.domain_object, self.monthly_plan_version, downgrades,
+        )
+        return downgrade_handler.get_response()
```

Comment:
Can we make this a method rather than a property? I like when properties adhere to property-like expectations -- i.e., lightweight, quick runtime. Specifically with the `downgrade_handler.get_response()` line, it looks like, in order for it to resolve something like the `CLOUDCARE` privileges, it needs to pull all a domain's application documents from Couch, which could be quite expensive.

One specific reason I've run into to avoid complicated properties is that the Visual Studio Code and PyCharm debuggers immediately evaluate properties they run into during breakpoints. That can cause unexpected behaviors or just general slowness.